python/identifiers.cpp

```diff
@@ -10,23 +10,31 @@
 namespace pyarb {
 
 using util::pprintf;
+namespace py = pybind11;
 
-void register_identifiers(pybind11::module& m) {
-    using namespace pybind11::literals;
+void register_identifiers(py::module& m) {
+    using namespace py::literals;
 
-    pybind11::class_<arb::cell_member_type> cell_member(m, "cell_member",
+    py::class_<arb::cell_member_type> cell_member(m, "cell_member",
         "For global identification of a cell-local item.\n\n"
         "Items of cell_member must:\n"
         "  (1) be associated with a unique cell, identified by the member gid;\n"
         "  (2) identify an item within a cell-local collection by the member index.\n");
 
     cell_member
-        .def(pybind11::init(
+        .def(py::init(
             [](arb::cell_gid_type gid, arb::cell_lid_type idx) {
                 return arb::cell_member_type{gid, idx};
             }),
             "gid"_a, "index"_a,
-            "Construct a cell member with arguments:\n"
+            "Construct a cell member identifier with arguments:\n"
+            "  gid:     The global identifier of the cell.\n"
+            "  index:   The cell-local index of the item.\n")
+        .def(py::init([](py::tuple t) {
+                if (py::len(t)!=2) throw std::runtime_error("tuple length != 4");
+                return arb::cell_member_type{t[0].cast<arb::cell_gid_type>(), t[1].cast<arb::cell_lid_type>()};
```

Comment:
What happens with these casts if the user passes a tuple with non-convertible types, e.g.:
```python
m = arbor.cell_member(("hello", 23.2))
```
You can force pybind11 to do the casting and checking for you by making the arguments to be `std::tuple<arb::cell_gid_type, arb::cell_lid_type>` instead of `pybind11::typle`.

Reply:
If we do that with the current version of the PR, we get:
```
RuntimeError: Unable to cast Python instance to C++ type (compile in debug mode for details)
```

Is that sufficient?

Reply:
Sufficient. I was hoping that it would just work! I always have to test.